rush.json

```diff
@@ -829,7 +829,8 @@
       "packageName": "heft-typescript-v4-test",
       "projectFolder": "build-tests/heft-typescript-v4-test",
       "reviewCategory": "tests",
-      "shouldPublish": false
+      "shouldPublish": false,
+      "decoupledLocalDependencies": ["@rushstack/eslint-config"]
```

Comment:
Is our eslint config dropping support for TS4? Can that be avoided? That should be noted in the changelog.

Reply:
No, `@typescript-eslint` still supposedly supports all the way back to 4.3.5:

[typescript-estree/src/parseSettings/warnAboutTSVersion.ts](https://github.com/typescript-eslint/typescript-eslint/blob/01556f50376b87eff21c42d121b627e3a6f003f0/packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts#L9)
```ts
/**
 * This needs to be kept in sync with /docs/users/Versioning.mdx
 * in the typescript-eslint monorepo
 */
const SUPPORTED_TYPESCRIPT_VERSIONS = '>=4.3.5 <5.4.0';
```

But there is some kind of Heft bug where ESLint and `heft-typescript-plugin` can be loading two different versions of the `typescript` engine, and if this happens then the `eslint` CLI will work fine but `heft` will fail with really strange lint errors such as:

```
Exception has occurred: TypeError: Cannot use 'in' operator to search for 'illegalDecorators' in 94
```

The problem is that a `enum` is getting created by Heft's TypeScript version but tested against an `enum` from ESLint's version, which causes an AST node to get misidentified, and then we're basically passing a totally wrong data structure into a utility function.

We should investigate this, but I didn't have time in this PR.  I expect the symptoms will go away after we upgrade the `decoupledLocalDependencies` (but show up again next time we do a nontrivial TypeScript upgrade).